src/components/panel/Panel.tsx

```diff
@@ -146,12 +155,14 @@ const Panel: FC = React.memo(() => {
           <div data-image-container className={`grow-0 shrink-0 ${showImage ? 'block' : 'hidden'} border-r border-border`}>
             {showImage && <ImageView />}
           </div>
-          <div data-scroll-container className={`h-full w-full bg-accent overflow-x-hidden overflow-y-auto relative`}>
-            <div data-text-container className={`bg-background p-3 pr-5 min-h-full relative flex border-r ${showSidebarBorders ? 'border-border' : 'border-transparent'} ${showTextOptions ? 'pt-16': ''} `}>
+          <div data-scroll-container className={`h-full w-full bg-accent relative ${annotMode === 'align' ? 'overflow-x-hidden overflow-y-auto' : '' }`}>
+            <div data-text-container className={`bg-background p-3 pr-5 min-h-full max-h-full  relative flex border-r
+                  ${showSidebarBorders ? 'border-border' : 'border-transparent'} ${showTextOptions ? 'pt-16': ''}
+                  ${annotMode === 'list' ? 'overflow-x-hidden overflow-y-auto' : ''} `}>
               {showText && <ErrorBoundary FallbackComponent={TextViewError}><TextView /></ErrorBoundary>}
             </div>
-            <div data-sidebar-container className={`absolute top-0 h-full w-[400px] px-2`}>
-              {showSidebarContent && <AnnotationsBody />}
+            <div data-sidebar-container className={`absolute top-0  max-h-full w-[400px] px-2 ${annotMode === 'list' ? 'overflow-x-hidden overflow-y-auto' : ''}`}>
+              {showSidebarContent && annotMode === 'align' ? <AnnotationsAlignMode /> : <AnnotationsListMode />}
```

Comment:
To declutter the Panel, maybe it would be good to wrap this conditional rendering into a separate component "AnnotationsRenderer"